### PR TITLE
Implement config reloading

### LINKED ISSUE #302: [feature request] reload config file on USR1 signal
I have found no way to reload the configuration without restarting the whole process (which is only possible through restarting i3). However, this is inconvenient and from time to time causes weird display bugs in my system.

### PR DESCRIPTION
**Pull request by MaxVerevkin**
Closes #302

### TODO
- [ ] Document this feature somewhere.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -223,8 +232,18 @@ fn run(matches: &ArgMatches) -> Result<()> {
                     },
                     signal_hook::consts::SIGUSR2 => {
                         //USR2 signal that should reload the config
-                        //TODO not implemented
-                        //unimplemented!("SIGUSR2 is meant to be used to reload the config toml, but this feature is yet not implemented");
+                        use std::ffi::CString;
+
+                        let exe = std::env::current_exe().unwrap();
```
Comment (ammgws) [inline src/main.rs:237]: Clever! That was the missing link

**Comment by ammgws** [comment]
Awesome stuff, it appears to work well - no zombies or anything. Possibly some corrupt files for blocks that write stuff to files but will just have to wait and see. 

Anyone from the original issue thread want to do some additional testing?
@faulesocke, @GladOSkar, @sharethewisdom, @boi4, @xxxserxxx, @Gelox

**Comment by MaxVerevkin** [comment]
> Possibly some corrupt files for blocks that write stuff to files

Do we have such blocks?

**Comment by ammgws** [comment]
Maybe the Pacman block when it updates the database. Or a Custom block. In any case there's not much we can do about it.

**Comment by GladOSkar** [comment]
Works perfectly for me

**Comment by ammgws** [comment]
Cheers :+1:

**Pull request merged**